### ISSUE TITLE
3 bugs I found while playing 1.7

### ISSUE THREAD

**Issue by Burger-Gaming** [KadeArchive/Kade-Engine#1937]
1. Even though the FPS is high, the notes are choppy.
2. During week 6, if you have CPU light strums turned on, the notes jitter.
3. If you pause during the song, one of two things happen:
Outcome 1: The game pauses as intended, and the song stops.
Outcome 2: The game stops all the animations as intended, loads the pause menu, but the music keeps going.

**Comment by nahiassis** [comment]
sustain note ends are bugged too

**Comment by Trayfellow** [comment]
i can confirm 1 and 3 happens to me too, 1 is really annoying cause im used to playing the game on 120fps

**Comment by Kade-github** [comment]
I cannot actually get choppy notes, I've fixed 3. and 2 is kinda smth I probably wont fix. but choppy notes I don't experience. what is your refresh rate?

**Comment by Trayfellow** [comment]
> I cannot actually get choppy notes, I've fixed 3. and 2 is kinda smth I probably wont fix. but choppy notes I don't experience. what is your refresh rate?

my monitors refresh rate is 120hz, and my fps cap is 120

**Comment by Kade-github** [comment]
that would explain it, my main monitor is at 60hz so I don't see this at all. I'll have to look more into this.

I guess a soft fix is to limit your monitor to 60hz while playing, but that's kind of a stupid solution.

**Comment by Trayfellow** [comment]
> that would explain it, my main monitor is at 60hz so I don't see this at all. I'll have to look more into this.
> 
> I guess a soft fix is to limit your monitor to 60hz while playing, but that's kind of a stupid solution.

that wont help, i play on 120fps since i have a smooth gameplay with it (and also because i got used to it lol)

**Comment by RagdollClash** [comment]
i also notice the choppy notes when running at 165hz

**Comment by nahiassis** [comment]
you guys didnt notice this?:
![image](https://user-images.githubusercontent.com/85742902/130598547-bb7584b0-e871-4c90-983a-7011cf3b0022.png)
here are my bug reports:
https://github.com/KadeDev/Kade-Engine/issues/1920
https://github.com/KadeDev/Kade-Engine/issues/1930

**Comment by Burger-Gaming** [comment]
> that would explain it, my main monitor is at 60hz so I don't see this at all. I'll have to look more into this.
> 
> I guess a soft fix is to limit your monitor to 60hz while playing, but that's kind of a stupid solution.

my main monitor is 60hz as well and i notice the choppy notes, even in FPS as high as 170

**Comment by RagdollClash** [comment]
incase it helps, here's a 120fps video i recorded from OBS. you can see the note scrolling is smooth until the song starts

https://user-images.githubusercontent.com/15317421/130830353-222bdd26-9a13-42b2-badf-67f155ba199e.mp4

**Comment by Burger-Gaming** [comment]
It's fairly hard to see lag in OBS, but that's basically the situation with bug 1

**Comment by Behnja** [comment]
> 
> 
> I cannot actually get choppy notes, I've fixed 3. and 2 is kinda smth I probably wont fix. but choppy notes I don't experience. what is your refresh rate?

can confirm choppy notes happen. I thought I was seeing things but then I played KadeEngine 1.6.2 and 1.7 to compare them and yup, there was a huge difference. 1.6.2's notes are smooth, 1.7's notes are choppy. I got a 144hz monitor, the difference was very noticeable.

**Comment by GhostSonic21** [comment]
Can also report the choppy notes thing. My monitor is 60hz, it happens even if I cap FPS to 60 or 120. This is running on an Nvidia RTX 2060 in case this is somehow a weird GPU-dependent issue.

Here's a video of 1.7 at 120fps cap recorded at 60fps:
https://youtu.be/jQA6OoMp_yA

And here's 1.6.2 at 120fps cap recorded at 60fps for comparison:
https://youtu.be/PDor6UBWQUc

**Comment by GhostSonic21** [comment]
Notes issue popped up in c442a64a2ad703ba9641165acf5a1803fdd063e8 , it's fine before then.

**Comment by RagdollClash** [comment]
seems like the note lag has been fixed with beceb21a3c386ff3a8d92511c514e42501e432db

also i think it has the additional bonus of dancing not going all crazy with the song speed changed, though it's still not synced properly

https://user-images.githubusercontent.com/15317421/131398653-b58f7bbb-a8e4-4c81-a702-c13be6917c43.mp4

**Comment by GhostSonic21** [comment]
Agreed, the choppy notes bug seems to be fixed now as of beceb21a3c386ff3a8d92511c514e42501e432db and b5010d4dda81487e53796008136eaa0def1e4b24 👍

**Comment by Behnja** [comment]
@ActualMandM I copied the code from [beceb21](https://github.com/KadeDev/Kade-Engine/commit/beceb21a3c386ff3a8d92511c514e42501e432db) and [b5010d4](https://github.com/KadeDev/Kade-Engine/commit/b5010d4dda81487e53796008136eaa0def1e4b24), and when I go to play a week the first song plays nicely, but when it's time to go on to the next song, the game crashes. Am I missing something? Is there some code from a past commit I should've copied?

**Comment by Behnja** [comment]
nvm just found what the issue was lmao
